### PR TITLE
Add wysiwyg

### PR DESCRIPTION
**Pull request by gridar**

### CODE REVIEW

File: app/javascript/src/locomotive/content_editing/inputs/text.jsx [hunk 1]
```diff
@@ -8,30 +13,119 @@ class TextInput extends Component {
...
+      html,
+      editorHeight,
+      line_break
+    };
+
+    this.inputOnChangeSanitizer = this.inputOnChangeSanitizer.bind(this);
+    this.editorOnChangeSanitizer = this.editorOnChangeSanitizer.bind(this);
+  }
+
+  formatLineBreak(text) {
```
Comment (did) [inline app/javascript/src/locomotive/content_editing/inputs/text.jsx]: put that in the utils/misc.js file

File: app/javascript/src/locomotive/content_editing/inputs/text.jsx [hunk 2]
```diff
@@ -8,30 +13,119 @@ class TextInput extends Component {
...
+    }
   }
 
-  handleChange(event) {
-    const { value } = event.target;
+  editorOnChangeSanitizer(editorState) {
+    this.setState({editorState: editorState});
+    let value = draftToHtml(convertToRaw(editorState.getCurrentContent()));
+
+    if(this.state.line_break) value = this.formatLineBreak(value)
```
Comment (did) [inline app/javascript/src/locomotive/content_editing/inputs/text.jsx]: space after if

File: app/javascript/src/locomotive/content_editing/inputs/text.jsx [hunk 3]
```diff
@@ -8,30 +13,119 @@ class TextInput extends Component {
...
 
-    this.state = { value };
+    //create new content with raw html
+    const { contentBlocks, entityMap } = htmlToDraft(value);
+    const contentState = ContentState.createFromBlockArray(contentBlocks, entityMap);
+    var editorState = EditorState.createWithContent(contentState);
+
+    var html = props.setting.html || false;
+
+    const line_height =  20;
```
[the remark this answers is not in the document]
Comment (gridar) [reply app/javascript/src/locomotive/content_editing/inputs/text.jsx]: done

File: app/javascript/src/locomotive/content_editing/inputs/text.jsx [hunk 4]
```diff
@@ -8,30 +13,119 @@ class TextInput extends Component {
...
+    var line_break = props.setting.line_break || false;
+
+    var editorHeight = (props.setting.rows || 5) * line_height;
+
+    this.state = {
+      editorState,
+      value,
+      html,
+      editorHeight,
```
[the remark this answers is not in the document]
Comment (gridar) [reply app/javascript/src/locomotive/content_editing/inputs/text.jsx]: done

File: app/javascript/src/locomotive/content_editing/inputs/text.jsx [hunk 5]
```diff
@@ -8,30 +13,119 @@ class TextInput extends Component {
...
+      line_break
```
[the remark this answers is not in the document]
Comment (gridar) [reply app/javascript/src/locomotive/content_editing/inputs/text.jsx]: done

File: app/javascript/src/locomotive/content_editing/inputs/text.jsx (hunk 1 above)
Comment (gridar) [reply app/javascript/src/locomotive/content_editing/inputs/text.jsx]: done

File: app/javascript/src/locomotive/content_editing/inputs/text.jsx [hunk 6]
```diff
@@ -8,30 +13,119 @@ class TextInput extends Component {
...
-
-    return (
-      <div className="editor-input editor-input-text">
-        <label>{setting.label}</label>
-        <br/>
-        <input type="text" value={this.state.value} onChange={this.handleChange} />
-      </div>
-    )
+    const { editorState } = this.state;
+    if(this.state.html) {
```
[the remark this answers is not in the document]
Comment (gridar) [reply app/javascript/src/locomotive/content_editing/inputs/text.jsx]: done

**Pull request merged**